contracts/Delegations.sol

```diff
@@ -148,56 +159,84 @@ contract Delegations is IDelegations, IStakeChangeNotifier, ContractRegistryAcce
 	function stakeMigration(address _stakeOwner, uint256 _amount) external onlyStakingContract {}
 
 	function _stakeChangeBatch(address[] memory _stakeOwners, uint256[] memory _amounts, bool[] memory _signs) private returns (address[] memory delegates){
-		uint256[] memory newUncappedStakes = new uint256[](_stakeOwners.length);
-		uint256[] memory prevGovStakeOwners = new uint256[](_stakeOwners.length);
-		uint256[] memory prevGovStakeDelegates = new uint256[](_stakeOwners.length);
+		if (_stakeOwners.length == 0) {
```

Comment:
I like this! 
Another approach is to keep _stakeChangeBatch like before, and do this optimization in the elections contract. I think this would require only a single pass so code may be a bit simpler. But I'm not sure which approach is better.

Reply:
after discussion - can remove first iteration over `_stakeOwners` - and delete `notifyStakeChangeBatch` from Elections if we only read `notifyStakeChange ` whenever a sequence closes.

Reply:
tested and saw packing and unpacking of multiple requests is much more expensive. keeping current structure